src/lib/HAPServiceNode.ts

```diff
@@ -165,8 +166,24 @@ module.exports = (RED: NodeAPI) => {
         // Service node properties
         self.name = self.config.name
 
+        // Find a unique identifier for the current service
+        let uniqueIdentifier
+        if(
+            self.hasOwnProperty('_flow') && 
+            self.hasOwnProperty('_alias') && 
+            self._flow.hasOwnProperty('TYPE') && 
+            FlowTypeType.Subflow == self._flow.TYPE
+        ) {
+            // For subflows, use the service node identifier from the subflow template 
+            // plus the full path from the subflow node identifier to the subflow.
+            uniqueIdentifier = self._alias + '/' + self._flow.path
+        } else {
+            // For top level flows, use the node identifier
+            uniqueIdentifier = self.id
+        }
+
         // Generate UUID from node id
```

Comment:
Line description to be changed or pruned as `node id` is no longer valid

Reply:
Thanks. I have always had a blind spot for comments ;-).